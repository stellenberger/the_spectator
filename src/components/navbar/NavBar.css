.navbar-container {
  background-color: gray;
  height: 180px;
}
