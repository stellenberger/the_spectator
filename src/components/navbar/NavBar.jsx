import React from 'react'
import './NavBar.css'

const NavBar = () => {
  return (
    <div data-testid='navbar' className='navbar-container'>
      <div data-testid='navbarTitle'>
        The Spectator
      </div> 
      This is the NavBar
    </div>
  )
}

export default NavBar