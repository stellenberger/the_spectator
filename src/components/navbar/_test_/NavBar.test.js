import React from 'react';
import ReactDOM from 'react-dom';
import NavBar from '../NavBar';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import renderer from 'react-test-renderer'
afterEach(cleanup);

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NavBar />, div)
})

it('renders navbarTitle correctly', () => {
  const { getByTestId } = render(<NavBar />)
  expect(getByTestId('navbarTitle').textContent).toBe('The Spectator')
})

it('matches snapshot', () => {
  const tree = renderer.create(<NavBar />).toJSON();
  expect(tree).toMatchSnapshot();
})
