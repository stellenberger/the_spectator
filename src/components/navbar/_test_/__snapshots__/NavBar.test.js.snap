// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="navbar-container"
  data-testid="navbar"
>
  <div
    data-testid="navbarTitle"
  >
    The Spectator
  </div>
  This is the NavBar
</div>
`;
